### ISSUE TITLE
History searching

### ISSUE THREAD

**Issue by luxaritas** [ircv3/ircv3-ideas#9]
Surprised this hasn't been brought up yet. It would be a logical extension of [CHATHISTORY](https://github.com/ircv3/ircv3-specifications/pull/292) from all I can tell. This is critical particularly to networks meant for intra-organization communication, where it may be important to reference a previous decision, explanation, etc.

**Comment by ghost** [comment]
I'd love a feature like this and thought about something similar when writing ircv3/ircv3-specifications#292. I decided to keep it out the the specification as it would break several expectations - most notably that the batch assumes chronological content from a single channel or query to be displayed in-line by the client. They would also need to be separate commands with a different set of parameters.

I think this would be better off as it's own specification, rather than an extension of ircv3/ircv3-specifications#292. With the exception of requiring some form of server-side logging, they do not share much else in terms of either form or function, although a `search` specification would certainly benefit from taking advantage of several of the same specifications, such as [batch](http://ircv3.net/specs/extensions/batch-3.2.html) and [labeled-response](http://ircv3.net/specs/extensions/labeled-response.html).

**Comment by luxaritas** [comment]
Right, and as I've actually gone back and read the spec I came to the same conclusion. I have some ideas for how this could potentially work.

**Comment by emersion** [comment]
Ref https://github.com/ircv3/ircv3-specifications/pull/496
